### ISSUE TITLE
 Article Comments User Images

### ISSUE THREAD

**Issue by sganz** [pencilblue/pencilblue#720]
![screenshot](https://cloud.githubusercontent.com/assets/5179047/9368207/cc562816-4676-11e5-9b44-cb88620a2c39.png)
Ran into this odd one. I had changed my user image prior to making any comments. I then made a comment and the old image showed up. Again, made another comment old image shows, then refreshed the page and now user images are as they should be.  Image shows first two comments then a page refresh then second set of comments with wrong (old) image.

I had also seen on a different install IF I only had the first user image when I made a comment on a post the user image shows as a broken link. On page refresh is fine. 

Firefox 40.0, PB 0.4.1

**Comment by blakecallens** [comment]
This sounds like browser caching. You said everything was fine after a page refresh?

**Comment by sganz** [comment]
Yes, but as I comment the wrong image is rendered. So the top 2 images are as I was adding comments, the bottom two were a result of the page refresh (they are correct). On a different VM where I was trying it when ever I added a new comment a broken image came up until I refreshed the page.  Are the URL's for the images unique? If so then the page may be generating the wrong url (not sure how the presentation layer caches) but I don't think it's a browser cache issue directly. I should have looked at the urls in firebug, will check next time.  The other issue that shows some odd behaviour is 'Comment (2)' which does not update until a full page refresh is done.

Thanks for the super quick reply, learning how it works from the outside in :)

Sandy

**Comment by sganz** [comment]
One quick note I look in firebug and after a restart of the VM and everything else I poked around and looked at the image URL's and they are different, so it may be caching something but it's not the browser image it seems, since PB is generating a different URL. So some caching may be going on but might be elsewhere then the browser.

**Comment by brianhyder** [comment]
@sganz Can you tell us a bit about your configuration and environment?  I'm mostly curious about what you are using as the media provider and clustering settings.  Would you mind confirming the behavior in a different browser?

**Comment by sganz** [comment]
Brian -

The environment starts off with a Win7 machine running a virtualbox with Mint 17.2 on it, and FF 40.0 as the browser. I'll try to get a look in another browser, but having some difficulty connecting to PB outside of the VBox for some reason (other web apps works fine). I might have a config issues with how the node binds to a port or something (new to node so...)

I'm using the out of the box PB install settings, here is my config.js -

```module.exports = {
    "siteName": "GTSparklplugs",
    "siteRoot": "http://localhost:8080",
    "sitePort": 8080,
    "logging": {
        "level": "info"
    },
    "db": {
        "type": "mongo",
        "servers": [
            "mongodb://127.0.0.1:27017/"
        ],
        "name": "pencilblue",
        "writeConcern": 1
    },
    "cache": {
        "fake": true,
        "host": "localhost",
        "port": 6379
    },
    "settings": {
        "use_memory": false,
        "use_cache": false
    },
    "templates": {
        "use_memory": true,
        "use_cache": false
    },
    "plugins": {
        "caching": {
            "use_memory": false,
            "use_cache": false
        }
    },
    "registry": {
        "type": "mongo"
    },
    "session": {
        "storage": "mongo"
    },
    "media": {
        "provider": "mongo",
        "max_upload_size": 6291456
    },
    "cluster": {
        "workers": 1,
        "self_managed": true
    },
    "siteIP": "127.0.0.1"
};
```

**Comment by brianhyder** [comment]
Awesome, thank you for the additional info.  I'll be able to try and reproduce this tonight and provide an update tonight or tomorrow morning.

**Comment by sganz** [comment]
I just tried it on IE from a win 10 box. Seems fine, went back to the VBox host and tried it again, fine.  Changed the icon for my account, added comments in VBox FF browser, fine. Did the same again in the Win10 browser and seems fine. 

That all being said this version of the code is the current master that I pulled after the merge that  fixed for the editor backspace issue. Not sure if anything in that branch might have affected the behaviour I saw.  I'll play with it some more but I'm not seeing the issue. I have another VM on another machine with it and will give it a try on that, that may have the old version still and if it does it on that, I'll pull the current master head and if it goes away problem solved.

I wouldn't waste much time on this now unless I see it again, I'm sure you have bigger fish to fry :)

Sandy
